### ISSUE TITLE
Server Secret Key Not Set

### ISSUE THREAD

**Issue by kingsykes** [robot-monkey/final-project#5]
If you are getting this error: ```TypeError: Secret string must be provided.```

then this means that the string used to generate a cryptographic signature has not been configured on your server.

Solution: from your project root navigate to /server/config/express.js
On or around line 30 should be 
```js
    app.use(session({
      secret: process.env.APP_SECRET,
      saveUninitialized: true,
      resave: true
    }));
```

change the ```secret``` property to any string, e.g. ```secret: 'nutz4bananas',```

If mocha is running  it should automatically detect the change and re-run the app.

If, when you ran grunt, mocha threw ```Error: Require key``` then this is also due to an unconfigured secret key. Although you can skip over it using the ```--force``` flag you can also find it and set it at /server/auth/auth.service.js in the variable declarations ```TOKEN_SECRET```

**Comment by kingsykes** [comment]
Alternatively, you can add these constants declarations to /node_modules/dotenv/.env and leave the other files as-is.
Declarations here will be loaded into the process.env object when the server is initialized.

**Comment by andrewolo** [comment]
How long do you think we'll need to piece it all together?

**Comment by cz2246** [comment]
Probably long enough to need an extension?

**Comment by joebeut** [comment]
Probably…

How are you guys doing?

I’m about done with the profiles — doing the catalog should be pretty easy after I finish. I have no idea how to merge stuff tho :/

**Comment by cz2246** [comment]
I am done with login app thing, i did it via the online course thing. Yet no idea how it will fit into the rest.

**Comment by andrewolo** [comment]
Yea prob -_-

I'm also linking up some of the html with the functions for editing of
products.

I figured data could just be passed in between the modules as they operate
"individually" and then we just style all the front end pages to look the
same. So each module would just have to define what input parameters it
needs.

**Comment by kingsykes** [comment]
Yeah I've tried to build the page and controller off of what's already there, but I realize now that this generator framework is really half-broken. I've almost got it done, but it's probably going to take more time than we have to hook up everything. At least I've spent a good amount of time chasing errors around so I have a decent idea how to put it all together.

**Comment by cz2246** [comment]
I have something working from the dreamspark online course thing. I can push that is lev's one is not working?

**Comment by andrewolo** [comment]
Push it please

**Comment by cz2246** [comment]
https://github.com/cz2246/project

Did it until authentication phase

**Comment by cz2246** [comment]
It is on my profile page on github

**Comment by cz2246** [comment]
Under repo called final project

**Comment by andrewolo** [comment]
Ok going to test yours out...

How do we go about addressing the extension with Prof?

**Comment by joebeut** [comment]
hey guys. ill push the profile management part soon. I have a midterm in the morning (9 am) so I dont know how much more I can get done tonight. I'll get the rest done tomorrow afternoon. Are we going to plan on submitting tomorrow night?

**Comment by cz2246** [comment]
Lets email him. I have never asked for one before though...

**Comment by cz2246** [comment]
Or should I write him? Even if we dont get the extension lets let them know we are not done. In morris' last email he said he would deduct 5 points for late

**Comment by andrewolo** [comment]
Yea could you email him please? 5 points is ok and we should have it done
today I believe right?

I'm working off your build in your profile so it should merge with yours
easily enough

**Comment by cz2246** [comment]
So sent it. No replies yet. Any updates on our end?

**Comment by cz2246** [comment]
So still no reply from Lev. How are we doing with the project?

**Comment by andrewolo** [comment]
My Web editor can now add and modify  products and have them display on the
page. Last step is to get it plugged in the data from mongodb. Addressing
that now.

So I'm about ready to merge with other parts if they can be pushed soon.

**Comment by cz2246** [comment]
Can you push it to the project github once done? Then whats the next step?

**Comment by andrewolo** [comment]
Yea I will. I had originally built mine to work off index.html and just
require the user to be logged in for the content to show. I can put it in a
separate module if necessary but I find this way to be most efficient.

If your authorization is working, we can probably at least of plug our 2
sections together

**Comment by cz2246** [comment]
Yeah my auth is working. But from hard coded mongo profile only

**Comment by andrewolo** [comment]
We have to get a server mongo working right?

**Comment by cz2246** [comment]
I did from server as well. You mean for huroku right?

**Comment by joebeut** [comment]
I have a server working locally. Do you have models built for the things you need to write for the schemas you write to the DB? i have profile and user schemas working for CRUD

**Comment by cz2246** [comment]
The code I pushed yesterday had a hard coded schema that works both locally and from huroku through a third party app. All we need to do is to change the schema i think

**Comment by joebeut** [comment]
I just pushed everything I have been working on to a new repo under robot-monkey/profile-user. Check it out. Im going to try to clean up the code to modularize it a bit more, but it should be clear enough to begin integrating. I used swig.js templates for the html working with express. I will try to move it to angular, but I don't know how to integrate it with everything else you guys have done.

**Comment by cz2246** [comment]
Ok. So next step is integrating right. Do we have anything remaining? Should we email lev again?

**Comment by joebeut** [comment]
I am supposed to do the catalog too, but did someone say they had built the
products already?

**Comment by andrewolo** [comment]
Well I got the manipulation of products going so we still need catalog.

If the catalog fits just like the format of your Add User Profile  then it
should work seamlessly

**Comment by joebeut** [comment]
ok. how are the products going to work with the catalog?

**Comment by andrewolo** [comment]
Based on the schema for products, a new product added would be inserted in
the catalog, and then we display to a user only the products that were
uploaded into the catalog and were not uploaded by that user right?

**Comment by cz2246** [comment]
I think so
